code/languages/c/static_memory_allocation/README.md

```diff
@@ -0,0 +1,27 @@
+## Static Memory Allocatiom
+It is allocated at compile time, and the lifetime of a variable in static memory is the lifetime of the program.
+
+In C, static memory can be allocated using the static keyword. The scope is the compilation unit only.
+
+Things get more interesting when the extern keyword is considered. When an extern variable is defined the compiler allocates memory for it. When an extern variable is declared, the compiler requires that the variable be defined elsewhere. Failure to declare/define extern variables will cause linking problems, while failure to declare/define static variables will cause compilation problems.
+
+in file scope, the static keyword is optional (outside of a function):
+``` c
+int a = 32;
+```
+But not in function scope (inside of a function):
+```c
+static int a = 32;
+```
+Technically, extern and static are two separate classes of variables in C.
+
+```c
+extern int a; /* Declaration */
+int a; /* Definition */
+```
+
+It's somewhat confusing to say that static memory is allocated at compile time, especially if we start considering that the compilation machine and the host machine might not be the same or might not even be on the same architecture.
```

Comment:
This text has been taken from https://stackoverflow.com/questions/8385322/difference-between-static-memory-allocation-and-dynamic-memory-allocation . Pls give proper credits.

Reply:
I have made the changes as told.